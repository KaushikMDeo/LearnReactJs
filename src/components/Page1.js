import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import Modal from './Modal';

const Page1 = React.createClass({
  propTypes: {
    children: PropTypes.node,
    location: PropTypes.object.isRequired
  },
  render() {
    let { location } = this.props;
    const isModal = location.state && location.state.modal;
    return (
      <div>
        <h1>Page 1</h1>
        <Link to={{
                pathname: `/page1/first`,
                state: { modal: true, returnTo: this.props.location.pathname }
              }}>First component</Link>
        {isModal && (
          <Modal title={'My First Component'} returnTo={location.state.returnTo}>
            {this.props.children}
          </Modal>
        )}
      </div>
    );
  }
});

export default Page1;
