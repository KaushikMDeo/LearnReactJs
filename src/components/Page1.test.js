import React from 'react';
import Page1 from './Page1';
import { shallow } from 'enzyme';
import serializer from 'enzyme-to-json/serializer';
expect.addSnapshotSerializer(serializer);

it('renders correctly', () => {
  const wrapper = shallow(
    <Page1 location={{}} />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders modal children correctly', () => {
  const wrapper = shallow(
    <Page1 location={{ state: { modal: true } }}>
      <h1>Test</h1>
    </Page1>
  );
  expect(wrapper).toMatchSnapshot();
});
