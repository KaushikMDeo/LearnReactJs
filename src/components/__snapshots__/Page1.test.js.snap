// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Page 1
  </h1>
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to={
      Object {
        "pathname": "/page1/first",
        "state": Object {
          "modal": true,
          "returnTo": undefined,
        },
      }
    }
  >
    First component
  </Link>
</div>
`;

exports[`renders modal children correctly 1`] = `
<div>
  <h1>
    Page 1
  </h1>
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to={
      Object {
        "pathname": "/page1/first",
        "state": Object {
          "modal": true,
          "returnTo": undefined,
        },
      }
    }
  >
    First component
  </Link>
  <withRouter(Modal)
    title="My First Component"
  >
    <h1>
      Test
    </h1>
  </withRouter(Modal)>
</div>
`;
